data_structures/src/chain.rs

```diff
@@ -2929,8 +2930,21 @@ fn compare_reputed_pkh(
     let rep_b = rep_eng.trs().get(b).0;
 
     rep_a.cmp(&rep_b).then_with(|| {
-        // FIXME(#1507): implement better tie breaker
-        a.cmp(b)
+        if rep_a > 0 {
+            let alpha_bytes = rep_eng.current_alpha.0.to_be_bytes().to_vec();
```

Comment:
You should be able to remove `.to_vec()` here if you set the type to slice:

```suggestion
            let alpha_bytes: &[u8] = &rep_eng.current_alpha.0.to_be_bytes();
```